Automattic/woocommerce-payments
# Set a locale when loading Stripe.js elements

Fixes #1752 

#### Changes proposed in this Pull Request

This PR adds the site locale when initializing Stripe.js elements, as a result the elements are rendered using the Site Language instead of the one used by the browser.

Stripe only supports a subset of the IETF language tags (https://stripe.com/docs/js/appendix/supported_locales), so a conversion is necessary to for example use `en` for `en_CA`, `en_AU` and `en_NZ`.

_NOTE_: It also changes the `handle` name for the register scripts calls to use the prefix `WCPAY_` needed to correctly load the javascript translations ([related PR](https://github.com/Automattic/woocommerce-payments/pull/1716))

#### Testing instructions
1. Change the site language to one that has a language pack available (and different than your browser locale)
2. Go to Dashboard > Updates, at the bottom of the page click: Update Translations

3. Go to the Shop, add any Product to the Cart, and proceed to the Classic Checkout page 
4. Select `Use a new payment method` and check that the placeholders in the Stripe fields are in the site language.

5. Go to the Shop, add any Product to the Cart, and proceed to the Blocks Checkout page
6. Check that the placeholders in the Stripe fields are in the site language (under `Payment method`)

-------------------
